### ISSUE TITLE
The requested platform 'SamsungTizen' was not registered by any plugin

### ISSUE THREAD

**Issue by KevinAPearson** [tavicu/homebridge-samsung-tizen#95]
Hi, I'm a total Noobie with my first Rapberry Pi 3+
I'm doing something wrong as I get this Error: "The requested platform 'SamsungTizen' was not registered by any plugin"
Here's what I get on screen:
[2019-4-28 11:50:42] Loaded config.json with 0 accessories and 1 platforms.
[2019-4-28 11:50:42] ---
[2019-4-28 11:50:42] No plugins found. See the README for information on installing plugins.
[2019-4-28 11:50:42] Loading 1 platforms...
[2019-4-28 11:50:42] Error: The requested platform 'SamsungTizen' was not registered by any plugin.
    at API.platform (/usr/local/lib/node_modules/homebridge/lib/api.js:134:13)
    at Server._loadPlatforms (/usr/local/lib/node_modules/homebridge/lib/server.js:327:45)
    at Server.run (/usr/local/lib/node_modules/homebridge/lib/server.js:90:36)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:59:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
[2019-4-28 11:50:42] Got SIGTERM, shutting down Homebridge...

This is my Config,Json text if that helps

{
        "bridge": {
                "name": "Homebridge",
                "username": "CC:22:3D:E3:CE:30",
                "port": 51826,
                "pin": "031-45-154"
        },

        "description": "This is an example configuration file with all supported devices. You can use this as a template f$

  "platforms": [{
      "platform": "SamsungTizen",
      "devices": [{
          "name": "Bedroom TV",
          "ip": "10.20.30.40",
          "mac": "A0:B1:C2:D3:E4:F5",
          "switches": [
              {"name": "Mute",      "mute": true},
              {"name": "Sleep",     "sleep": 60},
              {"name": "Channel",   "channel": 13},
              {"name": "Command 1", "command": "KEY_VOLUP"},
              {"name": "Command 2", "command": "KEY_VOLUP*5"},
              {"name": "Command 3", "command": ["KEY_VOLUP", "KEY_VOLDOWN"]},
              {"name": "Command 4", "command": ["KEY_VOLUP*5", "KEY_VOLDOWN*3"]},
              {"name": "All", "sleep": 60, "mute": true, "command": ["KEY_VOLUP", "KEY_VOLDOWN"], "channel": 13}
          ]
      }]
  }]
}

Thanks for any help.

**Comment by tavicu** [comment]
Hi. Can you please post the config file again but formatting as code. You can find more info how to do it here: https://help.github.com/en/articles/creating-and-highlighting-code-blocks

**Comment by tavicu** [comment]
But the error means that you dis not installed the plugin.

Follow the steps from here: https://github.com/tavicu/homebridge-samsung-tizen/blob/master/README.md#installation

**Comment by KevinAPearson** [comment]
Thanks for the help 
```
{
"bridge": {
"name": "Homebridge",
"username": "CC:22:3D:E3:CE:30",
"port": 51826,
"pin": "031-45-154"
},

    "description": "This is an example configuration file with all supported devices. You can use this as a template f$
"platforms": [{
"platform": "SamsungTizen",
"devices": [{
"name": "Bedroom TV",
"ip": "10.20.30.40",
"mac": "A0:B1:C2:D3:E4:F5",
"switches": [
{"name": "Mute", "mute": true},
{"name": "Sleep", "sleep": 60},
{"name": "Channel", "channel": 13},
{"name": "Command 1", "command": "KEY_VOLUP"},
{"name": "Command 2", "command": "KEY_VOLUP5"},
{"name": "Command 3", "command": ["KEY_VOLUP", "KEY_VOLDOWN"]},
{"name": "Command 4", "command": ["KEY_VOLUP5", "KEY_VOLDOWN*3"]},
{"name": "All", "sleep": 60, "mute": true, "command": ["KEY_VOLUP", "KEY_VOLDOWN"], "channel": 13}
]
}]
}]
}
```

**Comment by tavicu** [comment]
Follow the steps from here to install the plugin: https://github.com/tavicu/homebridge-samsung-tizen/blob/master/README.md#installation

**Comment by KevinAPearson** [comment]
Hi, I've reinstalled again but still get the same error
This is what I get:
pi@raspberrypi:~ $ sudo npm install -g --unsafe-perm homebridge-samsung-tizen
+ homebridge-samsung-tizen@3.1.0
added 57 packages in 14.722s
pi@raspberrypi:~ $ homebridge
[2019-4-28 12:05:44] Loaded config.json with 0 accessories and 1 platforms.
[2019-4-28 12:05:44] ---
[2019-4-28 12:05:44] ====================
[2019-4-28 12:05:44] ERROR LOADING PLUGIN homebridge-samsung-tizen:
[2019-4-28 12:05:44] /usr/local/lib/node_modules/homebridge-samsung-tizen/lib/de                   vice.js:18
            ...platform.config,
            ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge-samsung-tizen/                   index.js:2:15)
[2019-4-28 12:05:44] ====================
[2019-4-28 12:05:44] No plugins found. See the README for information on install                   ing plugins.
[2019-4-28 12:05:44] Loading 1 platforms...
[2019-4-28 12:05:44] Error: The requested platform 'SamsungTizen' was not regist                   ered by any plugin.
    at API.platform (/usr/local/lib/node_modules/homebridge/lib/api.js:134:13)
    at Server._loadPlatforms (/usr/local/lib/node_modules/homebridge/lib/server.                   js:327:45)
    at Server.run (/usr/local/lib/node_modules/homebridge/lib/server.js:90:36)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:59:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge                   :17:22)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
[2019-4-28 12:05:44] Got SIGTERM, shutting down Homebridge...
pi@raspberrypi:~ $

**Comment by tavicu** [comment]
Now you need to update your node to a newer version.

**Comment by KevinAPearson** [comment]
Thanks that sorted that problem.
Now it seems there's a problem with the config.json file:

```
{
	"bridge": {
		"name": "Homebridge",
		"username": "CC:22:3D:E3:CE:30",
		"port": 51826,
		"pin": "031-45-154"
	},

	"description": "This is an example configuration file with all supported devices. You can use this as a template for creating your own configuration file containing devices you actually own.",

  "platforms": [{
      "platform": "SamsungTizen",
      "devices": [{
          "name": "Bedroom TV",
          "ip": "192.168.0.26",
          "mac": "84:c0:ef:dd:57:cd",
          "switches": [
              {"name": "Mute",      "mute": true},
              {"name": "Sleep",     "sleep": 60},
              {"name": "Channel",   "channel": 13},
              {"name": "Command 1", "command": "KEY_VOLUP"},
              {"name": "Command 2", "command": "KEY_VOLUP*5"},
              {"name": "Command 3", "command": ["KEY_VOLUP", "KEY_VOLDOWN"]},
              {"name": "Command 4", "command": ["KEY_VOLUP*5", "KEY_VOLDOWN*3"]},
              {"name": "All", "sleep": 60, "mute": true, "command": ["KEY_VOLUP", "KEY_VOLDOWN"], "channel": 13}
          ]
      }]
  }]
```

**Comment by KevinAPearson** [comment]
Ok I get it I simply missed another } at the end to close it! (Doh!)
Thanks for all you help, I really appreciate it :-)

**Comment by tavicu** [comment]
Awesome :) glad i could help!

**Comment by KevinAPearson** [comment]
Thanks. All was working well but now I get:
 Failed to pair! Make sure TV is online and you click "Allow" on the popup
I originally did get the pop up but now the device is not listed in the TV list and the pop up won't appear!
Still having fun though! :-)

**Comment by KevinAPearson** [comment]
Just in case anyone comes across this problem:
I removed Homebridge only to discover it was still working after deleting it!
I guess I'd installed it badly in two locations (did I say I was a noobie?)
So I decided to take everything I'd learned with my first go at Rasp Pi and formatted the SD and started again. I ran everything in the correct way this time and now everything is working again. My Samsung TV asked for permission and as before and now everything is coming up roses :-)
Thanks to Tavicu for all their help and Good Luck to every one.
